pyuploadcare/api_resources.py

```diff
@@ -640,36 +631,66 @@ def __init__(self, **kwargs):
             setattr(self, f, kwargs.pop(f, default))
 
         if kwargs:
-            raise TypeError('Extra arguments: {}.'.format(
+            raise TypeError('Extra arguments: {0}.'.format(
                 ", ".join(kwargs.keys())
             ))
 
     def api_url(self, **additional):
         qs = {}
-        if self.since is not None:
-            qs['from'] = self.since.isoformat()
-        if self.until is not None:
-            qs['to'] = self.until.isoformat()
+
+        if self.starting_point is not None:
+            qs['from'] = self.starting_point
+
+        if self.ordering is not None:
+            qs['ordering'] = self.ordering
+
         if self.request_limit:
             qs['limit'] = self.request_limit
+
         for f, default in self.filters:
             v = getattr(self, f)
             if v == default:
                 continue
             qs[f] = 'none' if v is None else str(bool(v)).lower()
+
         qs.update(additional)
+
         return self.base_url + '?' + urlencode(qs)
 
+    @property
+    def starting_point(self):
+        """ Prepare value of the starting_point depending on the
+        current ordering.
+        """
+        if not self.__starting_point:
+            return None
+
+        ordering_field = (self.ordering or '').lstrip('-')
+        error_message = ('Incorrect value of the starting_point when '
+                         'ordering == {0}'.format(self.ordering))
+
+        if not ordering_field or ordering_field == 'datetime_uploaded':
+            try:
+                return dateutil.parser.parse(self.__starting_point).isoformat()
+            except ValueError:
+                raise ValueError(error_message)
+
+        if ordering_field == 'size':
+            try:
+                return int(self.__starting_point)
+            except (ValueError, TypeError):
+                raise ValueError(error_message)
+
+        raise ValueError('{0} is unknown ordering field'.format(self.ordering))
+
+    @starting_point.setter
+    def starting_point(self, value):
+        self.__starting_point = value
+
     def __iter__(self):
-        return api_iterator(
-            self.constructor, self.api_url(),
-            self.until is not None, self.limit,
-        )
+        return api_iterator(self.constructor, self.api_url(), self.limit)
 
     def count(self):
-        if self.since is not None or self.until is not None:
```

Comment:
This is obviously useful, but object counting is very heavy operation for database and can't be done with current infrastructure.